airflow/jobs.py

```diff
@@ -272,13 +272,29 @@ def manage_slas(self, dag, session=None):
             .filter(SlaMiss.dag_id == dag.dag_id)
             .all()
         )
+
+        current_run_date = datetime.now() - dag.schedule_interval
+        blocking_tis = (
+            session
+            .query(TI)
+            .filter(TI.state != State.SUCCESS)
+            .filter(TI.execution_date == current_run_date)
+            .filter(TI.are_dependencies_met())
```

Comment:
Should be good. No need to commit until values are changed and `session.merge` is called